engine/framework/worker.go

```diff
@@ -483,25 +494,33 @@ func (w *DefaultBaseWorker) OpenStorage(ctx context.Context, resourcePath resour
 }
 
 // Exit implements BaseWorker.Exit
-func (w *DefaultBaseWorker) Exit(ctx context.Context, status frameModel.WorkerStatus, err error) (errRet error) {
+func (w *DefaultBaseWorker) Exit(ctx context.Context, exitReason ExitReason, err error, errMsg string, extBytes []byte) (errRet error) {
```

Comment:
errMsg can be derived from err.Error. If err is nil, we should use extBytes

Reply:
I'm not so sure the `errMsg` need to be the same as `err.Error()` since `err.Error()` is less expressive.

Reply:
Keep the `errMsg` params to enable more concret error message

Reply:
there're 3 parameters `err error, errMsg string, extBytes []byte` but only 2 conpects. If developer wants to attach more concrete error message, he can add it to stacks or customer error fields